db/compaction/compaction_iterator.cc

```diff
@@ -271,12 +272,20 @@ void CompactionIterator::NextFromInput() {
 
     if (!ParseInternalKey(key_, &ikey_)) {
       iter_stats_.num_input_corrupt_records++;
+
       // If `expect_valid_internal_key_` is false, return the corrupted key
       // and let the caller decide what to do with it.
       // TODO(noetzli): We should have a more elegant solution for this.
       if (expect_valid_internal_key_) {
-        assert(!"Corrupted internal key not expected.");
-        status_ = Status::Corruption("Corrupted internal key not expected.");
+        if (log_corruption_errors_) {
+          std::string msg("Corrupted internal key not expected.");
```

Comment:
It's fine to log this part of the message even if the option is set to false since it doesn't contain user data.

Reply:
I was confused that it will anyway assert  if option is not set, so will that status be of any use in that case?

Reply:
I think yes because the `assert()`s are excluded from optimized builds. But it's good to delete this assertion anyways, as returning a `Status::Corruption` is the proper way to report corrupt data.